generator/src/main/java/com/scottlogic/deg/generator/generation/StandardFieldValueSourceEvaluator.java

```diff
@@ -8,133 +8,75 @@
 import java.util.*;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
-import java.util.stream.StreamSupport;
 
 public class StandardFieldValueSourceEvaluator implements FieldValueSourceEvaluator {
+    private static final CannedValuesFieldValueSource nullOnlySource = new CannedValuesFieldValueSource(Collections.singletonList(null));
+
     private final MustContainRestrictionReducer mustContainRestrictionReducer = new MustContainRestrictionReducer();
 
-    public Set<FieldValueSource> getFieldValueSources(FieldSpec fieldSpec){
-        Set<FieldValueSource> validSources = new HashSet<>();
-        MustContainRestriction mustContainRestriction = fieldSpec.getMustContainRestriction();
+    public List<FieldValueSource> getFieldValueSources(FieldSpec fieldSpec){
 
-        // check nullability...
-        if (determineNullabilityAndDecideWhetherToHalt(validSources, fieldSpec))
-            return validSources;
+        if (mustBeNull(fieldSpec)){
+            return Arrays.asList(nullOnlySource);
```

Comment:
Minor point, this could be expressed as (and would be suggested by intelliJ)
```suggestion
            return Collections.singletonList(nullOnlySource);
```